src/3rd_party_config.c

```diff
@@ -0,0 +1,173 @@
+#include "3rd_party_internal.h"
+#include "config.h"
+#include "lib/config_parser.h"
+#include "lib/list.h"
+#include "swupd.h"
+#include <stdlib.h>
+
+struct global {
+	void *value1;
+	void *value2;
+};
+
+struct repo {
+	char *repo_name;
+	char *url;
+};
+
+#define MAX_REPO_ALLOWED_FIELDS 2
+#define MAX_GLOBAL_ALLOWED_FIELDS 2
+
+struct list *repos_kv;
+
+bool parse_key_values(char *section, char *key, char *value, void *data)
+{
+	static int repo_field_count = 0;
+	static int global_field_count = 0;
+
+	if (!strncmp(section, "REPO", strlen(section))) {
+		if (key && value) {
+			static struct repo *repo_var;
+			if (repo_field_count == 0) {
+				repo_var = (struct repo *)malloc(sizeof(struct repo));
+			}
+			if (!strncmp(key, "repo-name", strlen(key))) {
+				repo_var->repo_name = strdup_or_die(value);
+				repo_field_count++;
+			}
+
+			if (!strncmp(key, "repo-url", strlen(key))) {
```

Comment:
I'm not sure about this approach because then you have to hardcode each one of the values of the config file to load one by one, if we want to add more values to the config file this parser would be broken, so it will be harder to maintain. For example, you are missing the `version` value for each repo which should be stored in the config file too.

Reply:
Also what is going to happen if you have multiple repos inside the repo.ini?